dbus/Cargo.toml

```diff
@@ -19,6 +19,7 @@ libdbus-sys = { path = "../libdbus-sys", version = "0.2.2" }
 futures-util = { version = "0.3", optional = true, default-features = false }
 futures-channel = { version = "0.3", optional = true }
 futures-executor = { version = "0.3", optional = true }
+arrayvec = { version = "0.7", optional = true }
 # dbus-native-channel = { path = "../dbus-native-channel", version = "0.1", optional = true }
```

Comment:
Could you give a motivation to your use of `arrayvec` (instead of using an ordinary `Vec`)? If it's just optimization, then I would argue it's premature optimization.

Reply:
Using an ordinary vec would not add any additional dependencies, meaning that we don't have to think about features. It just seems simpler to me.

Reply:
This was mostly done in preparation to *eventually* move things to a `const` context. As mentioned below there are a finite amount of rules that can actually be matched upon (and even less so in the current implementation) so this is an easy allocation to not do.

Reply:
Moving into that eventual const context is going to be difficult, if not impossible. The D-Bus library works with zero-terminated strings, and so that is the representation of paths, interfaces, etc in dbus-rs as well (even though if I ever finish off that native implementation it would make sense to have `&str` instead of `&CStr`). 

Which means that copying the strings once is almost inevitable, because you need to add that `\0` afterwards somehow.